mirgecom/eos.py

```diff
@@ -233,3 +232,236 @@ def total_energy(self, cv, pressure):
+    Pyrometheus <pyrometheus>` for underlying implementation details.
+
+    Each interface call expects that the :class:`mirgecom.euler.ConservedVars` object
+    representing the simulation conserved quantities contains at least the
+    canonical conserved quantities mass ($\rho$), energy ($\rho{E}$), and
+    momentum ($\rho\vec{V}$), and the vector of species masses, ($\rho{Y}_\alpha$).
+
+    .. important::
+        When using this EOS, users should take care to match solution initialization
+        with the appropriate units that are used in the user-provided `Cantera`
+        mechanism input files.
+
+    .. automethod:: __init__
+    .. automethod:: get_density
+    .. automethod:: get_species_molecular_weights
+    .. automethod:: get_production_rates
+    .. automethod:: get_species_source_terms
+    .. automethod:: get_internal_energy
+    .. automethod:: species_fractions
+    .. automethod:: total_energy
+    .. automethod:: gamma
+    .. automethod:: gas_const
+
+    Inherits from (and implements) :class:`GasEOS`.
+    """
+
+    def __init__(self, pyrometheus_mech, tguess=300.0):
+        """Initialize Pyrometheus-based EOS with mechanism class.
+
+        Parameters
+        ----------
+        pyrometheus_mech: :class:`pyrometheus.Thermochemistry`
+            The *Pyrometheus* mechanism :class:`~pyrometheus.Thermochemistry`
+            object that is generated by the user with a call to
+            *pyrometheus.get_thermochem_class*. To create the mechanism
+            object, users need to provide a mechanism input file. Several built-in
+            mechanisms are provided in `mirgecom/mechanisms/` and can be used through
+            the :meth:`mirgecom.mechanisms.get_mechanism_cti`.
+
+        tguess: float
+            This provides a constant starting temperature for the Newton iterations
+            used to find the mixture temperature. It defaults to 300.0 Kelvin. This
+            parameter is important for the performance and proper function of the
+            code. Users should set a tguess that is close to the average temperature
+            of the simulated domain.  Ideally, we would use the last computed
+            temperature for the guess, but doing so requires restart infrastructure
+            that is TBD.
+        """
+        self._pyrometheus_mech = pyrometheus_mech
+        self._tguess = tguess
+
+    def gamma(self, cv: ConservedVars = None):
+        r"""Get mixture-averaged specific heat ratio for mixture $\frac{C_p}{C_p - R_s}$.
+
+        Parameters
+        ----------
+        cv: :class:`mirgecom.euler.ConservedVars`
+            :class:`mirgecom.euler.ConservedVars` containing at least the mass
+            ($\rho$), energy ($\rho{E}$), momentum ($\rho\vec{V}$), and the vector of
+            species masses, ($\rho{Y}_\alpha$).
+        """
+        if cv is None:
+            raise NotImplementedError()
```

Comment:
I typically raise `NotImplementedError` for things that *could* be implemented but aren't. Here, it's my sense that this is impossible, so maybe `ValueError` is more appropriate. Also I prefer if there's an exception message, so that someone who runs into the error gets a bit of explanation.

Same below.

Reply:
Addressed in (117942b).